pkg/datapath/iptables/reconciler.go

```diff
@@ -143,15 +144,32 @@ stop:
 			}
 			state.localNodeInfo = localNodeInfo
 			stateChanged = true
-		case proxyInfo, ok := <-params.proxies:
+		case newProxyInfo, ok := <-params.proxies:
 			if !ok {
 				break stop
 			}
-			if state.proxies.Has(proxyInfo) {
+			if info, ok := state.proxies[newProxyInfo.name]; ok && info == newProxyInfo {
 				continue
 			}
-			state.proxies.Insert(proxyInfo)
-			stateChanged = true
+
+			// if existing, previous rules related to the previous entry for the same proxy name
+			// will be deleted by the manager (see Manager.addProxyRules)
+			state.proxies[newProxyInfo.name] = newProxyInfo
+
+			if !firstInit {
+				// first init not yet completed, proxy rules will be updated as part of that
+				stateChanged = true
+				continue
+			}
+
+			if err := updateProxyRules(newProxyInfo.port, newProxyInfo.isLocalOnly, newProxyInfo.name); err != nil {
+				log.WithError(err).Warning("iptables proxy rules incremental update failed, will retry a full reconciliation")
```

Comment:
Same issue as in the ticker.C batch case, this could cause a lot of sudden error logs.

Reply:
I've added rate limiting for errors logging in both full and partial reconciliation paths. I used two distinct rate limiters so that the partial errors (potentially very frequent) won't interfere with the more important full reconciliation error logs.